app/scripts/views/sign_up.js

```diff
@@ -217,14 +231,24 @@ define(function (require, exports, module) {
       var self = this;
       return p()
         .then(function () {
-          if (! self._isUserOldEnough()) {
-            self.notifier.trigger('signup.tooyoung');
-
-            return self._cannotCreateAccount();
+          if (self._isUserOldEnough()) {
+            self.notifier.trigger('signup.submit');
+            return self._initAccount(signUp);
           }
-          self.notifier.trigger('signup.submit');
 
-          return self._initAccount();
+          // COPPA is not valid, but maybe this is an existing user
+          // that wants to sign in. Let them try to sign in then, if
+          // that fails, come back here and show the COPPA error.
+          // https://github.com/mozilla/fxa-content-server/issues/2778
+          return self._initAccount(signIn)
+            .fail(function (err) {
+              if (self._coppa.hasValue()) {
```

Comment:
The failure section seems incorrect. There is no `onSignInError` method, so all sign in errors are handled here. Many errors can be returned from account.signIn, more than just the user is too young or failed to enter an age. It seems like there needs to be a branch that performs these checks only if the `err` is a certain value, probably `UNKNOWN_ACCOUNT`. As it stands, even network errors cause the user to hit one of these two conditions.